check/src/kindcheck.rs

```diff
@@ -112,9 +112,11 @@ impl<'a> KindCheck<'a> {
                 }
             },
                          Ok);
-        debug!("Find kind: {} => {}",
-               self.idents.string(&id),
-               kind.as_ref().unwrap());
+
+        if let Ok(ref kind) = kind {
+            debug!("Find kind: {} => {}", self.idents.string(&id), kind);
+        }
+
```

Comment:
Interestingly, `kind.as_ref().unwrap()` was causing a panic in the `undefined_type` test:

```
---- undefined_type stdout ----
	thread 'undefined_type' panicked at 'called `Result::unwrap()` on an `Err` value: Other(UndefinedType(0x10e60c088:Test))', ../src/libcore/result.rs:788
stack backtrace:
   1:        0x10d58fc3b - std::sys::backtrace::tracing::imp::write::h29f5fdb9fc0a7395
   2:        0x10d592eca - std::panicking::default_hook::_{{closure}}::h2cc84f0378700526
   3:        0x10d591989 - std::panicking::default_hook::hbbe7fa36a995aca0
   4:        0x10d5920c8 - std::panicking::rust_panic_with_hook::h105c3d42fcd2fb5e
   5:        0x10d591f26 - std::panicking::begin_panic::hbf62ea4a5ff3f9de
   6:        0x10d591e38 - std::panicking::begin_panic_fmt::h20f5943904e5791d
   7:        0x10d591d8f - rust_begin_unwind
   8:        0x10d5bcd80 - core::panicking::panic_fmt::h19323e466869c656
   9:        0x10d29b800 - core::result::unwrap_failed::h237e4f07e1529f12
  10:        0x10d28b54c - _<core..result..Result<T, E>>::unwrap::h1a70039526154d57
  11:        0x10d30d091 - gluon_check::kindcheck::KindCheck::find::h9fe9e43ed37f286d
  12:        0x10d30e29c - gluon_check::kindcheck::KindCheck::kindcheck::h4a4aed1b274f56fb
  13:        0x10d30d583 - gluon_check::kindcheck::KindCheck::kindcheck_type::h11a2e5fec88624b4
  14:        0x10d2f03bb - gluon_check::typecheck::Typecheck::typecheck_::hbf109e3fee47828e
  15:        0x10d2e7e51 - gluon_check::typecheck::Typecheck::typecheck::hbb9201e353574974
  16:        0x10d2e730a - gluon_check::typecheck::Typecheck::typecheck_expr_expected::hbf06a30d3dc297ef
  17:        0x10d2e7275 - gluon_check::typecheck::Typecheck::typecheck_expr::h283d7a8657c7ad0a
  18:        0x10d20bd14 - typecheck::functions::typecheck_expr::h092b5baa80b31d27
  19:        0x10d20b3ee - typecheck::functions::typecheck::h6793b14bc033042a
  20:        0x10d215985 - typecheck::undefined_type::ha48cdff39e7b53c0
  21:        0x10d2289ab - _<F as alloc..boxed..FnBox<A>>::call_box::h6c46b1f51a5f97cd
  22:        0x10d23d31a - test::run_test::run_test_inner::_{{closure}}::_{{closure}}::h264d6ff770d7074c
  23:        0x10d22068d - std::panicking::try::call::h8d6a00e6c0ac8b9e
  24:        0x10d5931eb - __rust_try
  25:        0x10d5930a5 - __rust_maybe_catch_panic
  26:        0x10d23b065 - std::thread::Builder::spawn::_{{closure}}::h9e5d89388555e457
  27:        0x10d228758 - _<F as alloc..boxed..FnBox<A>>::call_box::h173484225f9723d2
  28:        0x10d590f98 - std::sys::thread::Thread::new::thread_start::h8f3bd45211e9f5ea
  29:     0x7fff8a20899c - _pthread_body
  30:     0x7fff8a208919 - _pthread_start
```

Maybe that highlights a problem?